homeassistant/components/hyperion/light.py

```diff
@@ -360,11 +339,10 @@ def effect(self) -> str:
     @property
     def effect_list(self) -> List[str]:
         """Return the list of supported effects."""
-        return (
-            self._effect_list
-            + list(const.KEY_COMPONENTID_EXTERNAL_SOURCES)
-            + [KEY_EFFECT_SOLID]
-        )
+        effects = self._effect_list + [KEY_EFFECT_SOLID]
+        if self._support_external_effects:
+            effects += list(const.KEY_COMPONENTID_EXTERNAL_SOURCES)
```

Comment:
Is this something you could pre compute in the constructor or inside `async_added_to_hass` instead of each time this property is fetched ?